### PR TITLE
ops/terrad: update InitOCR

### PR DESCRIPTION
**Pull request by jmank88**
https://app.shortcut.com/chainlinklabs/story/29678

### CODE REVIEW

File: ops/terrad/terrad.go [hunk 1]
```diff
@@ -353,28 +371,162 @@ func (t Terrad) InitOCR(keys []opsChainlink.NodeKeys) error {
...
+					id = a.Value
+				}
+			}
+		}
+	}
+	if id == "" {
+		return errors.New("failed to find event with attribute: wasm.proposal-id")
+	}
+
+	defer func() {
```
Comment (krebernisak) [inline ops/terrad/terrad.go:404]: Proposals are cleared after being accepted, so not really necessary unless the idea is to clear broken proposals on error. In that case, an additional comment would be appreciated, about why the code is here.

File: ops/terrad/terrad.go [hunk 2]
```diff
@@ -353,28 +371,162 @@ func (t Terrad) InitOCR(keys []opsChainlink.NodeKeys) error {
...
+			F:             f,
+			OnchainConfig: onchainConfig,
+		},
+	})
+	if err == nil && resp.Code != 0 {
+		err = fmt.Errorf("tx response contains error: %s %d", resp.Codespace, resp.Code)
+	}
+	if status.Check(err) != nil {
+		return err
+	}
```
Comment (krebernisak) [inline ops/terrad/terrad.go:444]: Potentially we could shed some repeated code by extracting this as:

```suggestion
	if err = t.Check(err); err != nil {
		return err
	}
```

File: ops/terrad/terrad.go [hunk 3]
```diff
@@ -353,28 +371,162 @@ func (t Terrad) InitOCR(keys []opsChainlink.NodeKeys) error {
...
+
+	return nil
+}
+
+func (t Terrad) executeOCR2(msg interface{}) (resp TxResponse, err error) {
```
Comment (krebernisak) [inline ops/terrad/terrad.go:512]:
```suggestion
func (t Terrad) Execute(addr string, msg interface{}) (resp TxResponse, err error) {
```

**Review by krebernisak — CHANGES_REQUESTED** [review]

File: ops/terrad/terrad.go (hunk 2 above)
Comment (jmank88) [reply ops/terrad/terrad.go:444]: :+1: although I think pushing it inside of the execute method is even simpler.

File: ops/go.mod [hunk 4]
```diff
@@ -243,3 +243,6 @@ require (
 
 // Usage of this gogo/protobuf fork is required by Cosmos SDK: https://github.com/cosmos/cosmos-sdk/issues/10702
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+
+// Notice: uncomment while developing
+// replace github.com/smartcontractkit/chainlink-relay/ops => ../../chainlink-relay/ops
```
Comment (RyanRHall) [inline ops/go.mod:248]: unsure about how helpful this is to leave in

File: ops/Pulumi.localterra.yaml [hunk 5]
```diff
@@ -5,36 +5,47 @@ config:
...
   terra-env:CL-PORT-START: 6688 # start port to expose lite client port
   terra-env:CL-P2P_PORT-START: 5050 # start port to expose lite client P2P ports
-  # terra-env:CL-NODE_VERSION: develop.242e63a8dc8c9b1b7d5424a81e4ff7eb1461b3c0 # development version can be found at https://gallery.ecr.aws/chainlink/chainlink
-  terra-env:CL-BUILD_LOCALLY: true # (overrides NODE_VERSION) BUILD_CONTEXT and BUILD_DOCKERFILE must be set if this is enabled
-  terra-env:CL-BUILD_CONTEXT: ../../chainlink
-  terra-env:CL-BUILD_DOCKERFILE: ../../chainlink/core/chainlink.Dockerfile
+  terra-env:CL-NODE_VERSION: develop.d9ee81244df2417a80de5d83f8692a9c8be1e0e5 # development version can be found at https://gallery.ecr.aws/chainlink/chainlink
+  terra-env:CL-BUILD_LOCALLY: false # (overrides NODE_VERSION) BUILD_CONTEXT and BUILD_DOCKERFILE which must be set if this is enabled
+  # terra-env:CL-BUILD_CONTEXT: ../../chainlink
+  # terra-env:CL-BUILD_DOCKERFILE: ../../chainlink/core/chainlink.Dockerfile
```
[the remark this answers is not in the document]
Comment (jmank88) [reply ops/Pulumi.localterra.yaml:11]: @krebernisak was this commit meant only to bump `github.com/smartcontractkit/chainlink-relay/ops`?

[the remark this answers is not in the document]
Comment (krebernisak) [reply ops/Pulumi.localterra.yaml:11]: Update the `github.com/smartcontractkit/chainlink-relay/ops` and `github.com/smartcontractkit/chainlink-relay/ops` as required by the bump and bringing parity with Solana.

I actually don't prefer having `CL-BUILD_LOCALLY: true` as this entails more complex setup for example to run it in the CI, which is something we should add soon so local env doesn't keep breaking.

File: ops/go.mod (hunk 4 above)
Comment (krebernisak) [reply ops/go.mod:248]: Yeah, I can remove it. Could be helpful to have a quick local setting as we do interact x-repo a lot (chainlink, chainlink-relay, integrations-framework), but maybe a better place would be README.

File: ops/Pulumi.localterra.yaml [hunk 6]
```diff
@@ -5,36 +5,47 @@ config:
   terra-env:CL-COUNT: 4 # not counting bootstrap, minimum 4
   terra-env:CL-PORT-START: 6688 # start port to expose lite client port
   terra-env:CL-P2P_PORT-START: 5050 # start port to expose lite client P2P ports
-  # terra-env:CL-NODE_VERSION: develop.242e63a8dc8c9b1b7d5424a81e4ff7eb1461b3c0 # development version can be found at https://gallery.ecr.aws/chainlink/chainlink
-  terra-env:CL-BUILD_LOCALLY: true # (overrides NODE_VERSION) BUILD_CONTEXT and BUILD_DOCKERFILE must be set if this is enabled
-  terra-env:CL-BUILD_CONTEXT: ../../chainlink
-  terra-env:CL-BUILD_DOCKERFILE: ../../chainlink/core/chainlink.Dockerfile
+  terra-env:CL-NODE_VERSION: develop.63b7b698e1e22f334dfbcc35e0a875861da399fa # development version can be found at https://gallery.ecr.aws/chainlink/chainlink
```
Comment (krebernisak) [inline ops/Pulumi.localterra.yaml:8]:
```suggestion
  terra-env:CL-NODE_VERSION: develop.27b4494ba0b33178333908d72db6689ee9d8902d # development version can be found at https://gallery.ecr.aws/chainlink/chainlink
```

**Review by krebernisak — APPROVED** [review]

**Pull request merged**